peercred_freebsd.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2021 AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package peercred
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+
+	"golang.org/x/sys/unix"
+)
+
+func init() {
+	osGet = getFreeBSD
+}
+
+func getFreeBSD(c net.Conn) (*Creds, error) {
+	switch c := c.(type) {
+	case *net.UnixConn:
+		return getUnix(c)
+	case *net.TCPConn:
+		// TODO: use /proc tcp info for localhost connections like Windows?
```

Comment:
does FreeBSD have a /proc/net/tcp like Linux? I don't know how netstat works on BSD.

Reply:
Updated to refer to sysctl.